core/api/src/servers/ws-server.ts

```diff
@@ -69,12 +69,13 @@ const getContext = async (
 
       const ip = parseIps(ipString)
 
+      const apiKey = connectionParams?.["X-API-KEY"] as AuthToken | undefined
       const authz = connectionParams?.Authorization as string | undefined
-      const kratosToken = authz?.slice(7) as AuthToken
+      const bearerToken = authz?.slice(7) as AuthToken
```

Comment:
maybe for another PR but we should probably not use the same type for different type of token. probably apiKey should be `ApiToken` type